Google.PowerShell/Common/ActiveUserToken.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+
+namespace Google.PowerShell.Common
+{
+    /// <summary>
+    /// OAuth 2.0 model for a successful access token response as specified in 
+    /// http://tools.ietf.org/html/rfc6749#section-5.1.
+    /// </summary>
+    public class ActiveUserToken
+    {
+        /// <summary>
+        /// The user that this token corresponds to.
+        /// </summary>
+        public string User;
+
+        /// <summary>The access token issued by the authorization server.</summary>
+        public string AccessToken { get; internal set; }
+
+        /// <summary>
+        /// The date and time that this token was issued.
+        /// This is set in UTC time.
+        /// It should be set by the CLIENT after the token was received from the server.
+        /// </summary>
+        public DateTime ExpiredTime { get; internal set; }
+
+        /// <summary>
+        /// Returns <c>true</c> if the token is expired or it's going to be expired in the next minute.
+        /// </summary>
+        public bool IsExpired
+        {
+            get
+            {
+                if (AccessToken == null)
+                {
+                    return true;
+                }
+
+                if (ExpiredTime == DateTime.MaxValue)
+                {
+                    return false;
+                }
+
+                return ExpiredTime.AddSeconds(-60) <= DateTime.UtcNow;
+            }
+        }
+
+        /// <summary>
+        /// Constructs a new token by parsing userCredentialJson.
+        /// User will be set to the current active user.
+        /// </summary>
+        public ActiveUserToken(string userCredentialJson) :
```

Comment:
Given that this constructor is only used in one place, and it coules this type with the `CloudSdkSettings`, why don't we remove this. And force callers to specify the user.